src/Npgsql/Replication/ReplicationConnection.cs

```diff
@@ -452,8 +452,11 @@ internal async IAsyncEnumerator<XLogDataMessage> StartReplicationInternal(
 
             SetTimeouts(_walReceiverTimeout, CommandTimeout);
 
-            _sendFeedbackTimer = new Timer(TimerSendFeedback, state: null, WalReceiverStatusInterval, Timeout.InfiniteTimeSpan);
-            _requestFeedbackTimer = new Timer(TimerRequestFeedback, state: null, _requestFeedbackInterval, Timeout.InfiniteTimeSpan);
+            using (ExecutionContext.SuppressFlow()) // Don't capture the current ExecutionContext and its AsyncLocals onto the timer causing them to live forever
```

Comment:
Strictly speaking, this shouldn't really be necessary since this is `IAsyncEnumerator` and timers will be disposed whenever the enumerator is, but might as well do this for consistency.